src/backend/marsha/core/api/live_session.py

```diff
@@ -299,3 +301,32 @@ def set_display_name(self, request):
                 )
 
             raise error
+
+    @method_decorator(cache_page(settings.VIDEO_ATTENDANCES_CACHE_DURATION))
+    @action(
+        detail=False,
+        methods=["get"],
+        permission_classes=[permissions.IsTokenInstructor | permissions.IsTokenAdmin],
+        url_path="list_attendances",
+    )
+    def list_attendances(self, request, *args, **kwargs):
+        """
+        Retrieve the list of attendances computed for livesessions where is_registered
+        equals True or with the live_attendance field not empty. As we request a list
+        of statistics, we want the timeline of the video to be identical for all
+        students, we force the cache reinitialization for the timeline of the video.
+        The same one will then be used for all the students on this list.
+        """
+
+        Video.objects.get(pk=self.request.user.id).get_list_timestamps_attendences(
+            force=True
+        )
```

Comment:
```suggestion
```

Should not be removed ? Otherwise there is no interest to have a cache in the model ?

Reply:
The comments explains why this is done, the view is cached on top, so it won't get recalculated if the listing has already been queried

Reply:
What do you suggest then, cause I had already did a lot of thinking about this.
IMO if the timeline changes for the page 2 it doesn't really matter, it would still be correct, all the points will be pointing at the same time, my problem was for it to change on one query.

Reply:
That wouldn't solve my worries, what do you think to keep the key if it exists and extend it that would answer to both of the cases and it should make us both happy